pkg/che/che.go

```diff
@@ -0,0 +1,48 @@
+package che
+
+import (
+	"context"
+	"fmt"
+
+	crtConfig "github.com/codeready-toolchain/member-operator/pkg/configuration"
+
+	routev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("che-client")
+
+// DefaultClient is a default implementation of a CheClient
+var DefaultClient *Client
+
+// Client is a client for interacting with Che services
+type Client struct {
+	config     *crtConfig.Config
+	k8sClient  client.Client
+	tokenCache *tokenCache
+}
+
+// InitDefaultCheClient initializes the default Che service instance
+func InitDefaultCheClient(cfg *crtConfig.Config, cl client.Client) {
+	DefaultClient = &Client{
+		config:     cfg,
+		k8sClient:  cl,
+		tokenCache: newTokenCache(),
+	}
+}
+
+func getRouteURL(cl client.Client, namespace, name string) (string, error) {
```

Comment:
In addition, it looks like a copy of https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/memberstatus/memberstatus_controller.go#L375-L387 - wouldn't it be good to export it somewhere and reuse it so we don't have to maintain it at two places?

Reply:
I'll be using it for che API calls in che.go as well but I agree it can be a common function somewhere. Any suggestions where? Maybe a `utils` package? eg. `github.com/codeready-toolchain/member-operator/pkg/utils` with a `route.go` file?